documentation/ha.markdown

```diff
@@ -0,0 +1,265 @@
+---
+title: "PuppetDB 4.2: PuppetDB HA"
+layout: default
+canonical: "/puppetdb/latest/ha.html
+---
+
+[configure]: ./configure.html
+[pe-configure]: ./pe-configure.html
+[logging]: ./logging.html
+[metrics]: ./api/metrics/v1/index.html
+
+Configuring PuppetDB for High Availability
+-----
+
+In Puppet Enterprise 2016.5 and later, PuppetDB may be configured for high
+availability in order to withstand network partitions or system failure.
+
+PuppetDB is automatically configured for high availability as part of an HA
+deployment of Puppet Enterprise. For more information about high availability in
+Puppet Enterprise, see [High availability overview]({{pe}}/ha_overview.html).
+
+HA Overview
+-----
+
+PuppetDB HA has two parts: first, Puppet Server is configured to send commands
+and queries to multiple PuppetDB servers. Second, those servers are configured
+to periodically reconcile their differences, transferring any records which they
+are missing. This process is pull-based and runs on a configurable interval.
+
+PuppetDB replication is in principle a multi-master system. You can issue
+commands to any server, and they can be reconciled with any other server without
+conflicts. But, in order to minimize confusion when using exported resources, we
+recommend that only node be conventionally treated as primary.
+([See below](#exportedResources) for further discussion)
+
+PuppetDB HA is available in Puppet Enterprise 2016.5 and later.
+
+Manual Installation and Configuration
+----
+
+1. Provision two PuppetDB nodes. Designate one of these as your _primary_
+   PuppetDB, and the other as the _replica_. Be sure that each PuppetDB is using
+   its own PostgreSQL cluster. HA doesn't buy you anything if you have single
+   database. In this guide, we will give them the hostnames
+   `primary-puppetdb.mycorp.net` and `replica-puppetdb.mycorp.net`.
+
+2. Configure each PuppetDB to pull from the other. This guide will use
+   HOCON-style configuration. See the
+   [pe-configure][PE Configuration Documentation] for further details.
+
+  a. Configure primary-puppetdb to pull from replica-puppetdb by placing this in its
+     config file:
+
+         sync: {
+           remotes: [{server_url: "https://replica-puppetdb.mycorp.net:8081",
+                      interval: 5m}]
+         }
+
+  b. Configure replica-puppetdb to pull from primary-puppetdb:
+
+         sync: {
+           remotes: [{server_url: "https://primary-puppetdb.mycorp.net:8081",
+                      interval: 5m}]
+         }
+
+3. Configure the Puppet master to point to both of your PuppetDB instances. In
+   your puppetdb.conf, use this as your server configuration (in the 'main' section):
+
+        server_urls: ["https://primary-puppetdb.mycorp.net:8081", "https://replica-puppetdb.mycorp.net:8081"]
```

Comment:
Are we actually specifying puppetdb.conf in hocon? PE davis (at least) ships with an ini formatted file, which would mean the colons should be equal signs and the array should be a comma-separated string.